standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/properties/PropertyManager.java

```diff
@@ -0,0 +1,576 @@
+/**
+ * A property manager.
+ * <p>
+ * This handles operations at the higher functional level; an instance is created per-session and
+ * drives queries and updates in a transactional manner.
+ * </p>
+ * <p>
+ * The manager ties the property schemas into one namespace; all property maps it handles must and will use
+ * one of its known schema.
+ * </p>
+ */
+public abstract class PropertyManager {
+  /** The logger. */
+  public static final Logger LOGGER = LoggerFactory.getLogger(PropertyManager.class);
+  /** The set of dirty maps. */
+  protected Map<String, PropertyMap> dirtyMaps = new HashMap<>();
+  /** This manager namespace. */
+  protected final String namespace;
+  /** The property map store. */
+  protected final PropertyStore store;
+  /** A Jexl engine for convenience. */
+  static final JexlEngine JEXL;
+  static {
+    JexlFeatures features = new JexlFeatures()
+        .sideEffect(false)
+        .sideEffectGlobal(false);
+    JexlPermissions p = JexlPermissions.RESTRICTED
+        .compose("org.apache.hadoop.hive.metastore.properties.*");
+    JEXL = new JexlBuilder()
+        .features(features)
+        .permissions(p)
+        .create();
+  }
+
+  /**
+   * The map of defined managers.
+   */
+  private static final Map<String, Constructor<? extends PropertyManager>> NSMANAGERS = new HashMap<>();
+
+  /**
+   * Declares a property manager class.
+   * @param ns the namespace
+   * @param pmClazz the property mamanger class
+   */
+  public static boolean declare(String ns, Class<? extends PropertyManager> pmClazz) {
+    try {
+      synchronized(NSMANAGERS) {
+        Constructor<? extends PropertyManager> ctor = NSMANAGERS.get(ns);
+        if (ctor == null) {
+          ctor = pmClazz.getConstructor(String.class, PropertyStore.class);
+          NSMANAGERS.put(ns, ctor);
+          return true;
+        } else {
+          if (!Objects.equals(ctor.getDeclaringClass(), pmClazz)) {
+            LOGGER.error("namespace {} is already declared for {}", ns, pmClazz.getCanonicalName());
+          }
+        }
+      }
+    } catch(NoSuchMethodException xnom ) {
+      LOGGER.error("namespace declaration failed: " + ns + ", " + pmClazz.getCanonicalName(),
+          xnom);
+    }
+    return false;
+  }
+
+  /**
+   * Creates an instance of manager using its declared namespace.
+   * @param namespace the manager&quot;s namespace
+   * @param store the property store
+   * @return a property manager instance
+   */
+  public static PropertyManager create(String namespace, PropertyStore store) {
+    final Constructor<? extends PropertyManager> ctor;
+    synchronized(NSMANAGERS) {
+      ctor = NSMANAGERS.get(namespace);
+    }
+    if (ctor != null) {
+      try {
+        return ctor.newInstance(namespace, store);
+      } catch(Exception xany) {
+        LOGGER.error("property manager creation failed "+ namespace, xany);
+      }
+    } else {
+      LOGGER.error("no such property manager namespace is declared " + namespace);
```

Comment:
An error is not silent :-) ; it also should only occur during development of a new PropertyManager since it means the constructor failed. I'm trying to avoid throwing an exception that could crash the server or avoid it to start.

Reply:
Made it throw errors.